cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -282,6 +286,18 @@ def route_spec_matches_route(route_spec, route):
         'interface_id': 'interface_id',
         'vpc_peering_connection_id': 'vpc_peering_connection_id',
     }
+
+    # This is a workaround to catch managed NAT gateways as they do not show
+    # up in any of the returned values when describing route tables.
+    # The caveat of doing it this way is that if there was an existing
+    # route for another nat gateway in this route table there is not a way to
+    # change to another nat gateway id. Long term solution would be to utilise
+    # boto3 which is a very big task for this module or to update boto.
+    if route_spec.get('gateway_id') and 'nat-' in route_spec['gateway_id']:
+        if route.destination_cidr_block == route_spec['destination_cidr_block']:
+            if all((not route.gateway_id, not route.instance_id, not route.interface_id, not route.vpc_peering_connection_id)):
```

Comment:
Hi @erydo. 

I believe if I was to change it to if no any ((<items>)) this would not perform the action required to catch managed nat gateways. the if all (not blah, not blah) is to ensure it checks that none of the fields are not present which would then at this stage mean it was a managed nat gateway.